File: src/app/src/main/java/com/onscreensync/tvapp/apiresponses/configs/DisplayApiConfig.java
package com.onscreensync.tvapp.apiresponses.configs;

import com.google.gson.annotations.SerializedName;

public class DisplayApiConfig {
    @SerializedName("base-endpoint")
    private String baseEndpoint;
    @SerializedName("device-code-url")
    private String deviceCodeUrl;
    @SerializedName("device-info-url")
    private String deviceInfoUrl;
    @SerializedName("device-token-request-url")
    private String deviceTokenRequestUrl;
    @SerializedName("device-refresh-token-request-url")
    private String deviceRefreshTokenRequestUrl;
    @SerializedName("content-data-url")
    private String contentDataUrl;
    @SerializedName("signalr-negotiation-url")
    private String signalrNegotiationUrl;
    @SerializedName("signalr-add-connection-url")
    private String signalrAddConnectionUrl;
    @SerializedName("signalr-remove-connection-url")
    private String signalrRemoveConnectionUrl;
    @SerializedName("messages")
    private DisplayApiConfigMessages configMessages;

    public String getBaseEndpoint() {
        return baseEndpoint;
    }

    public String getDeviceCodeUrl() {
        return deviceCodeUrl;
    }

    public String getDeviceInfoUrl() {
        return deviceInfoUrl;
    }

    public String getDeviceTokenRequestUrl() {
        return deviceTokenRequestUrl;
    }

    public String getDeviceRefreshTokenRequestUrl() {
        return deviceRefreshTokenRequestUrl;
    }

    public String getContentDataUrl() {
        return contentDataUrl;
    }

    public String getSignalrNegotiationUrl() {
        return signalrNegotiationUrl;
    }

    public String getSignalrAddConnectionUrl() {
        return signalrAddConnectionUrl;
    }

    public String getSignalrRemoveConnectionUrl() {
        return signalrRemoveConnectionUrl;
    }

    public DisplayApiConfigMessages getConfigMessages() {
        return configMessages;
    }
}

File: src/app/src/main/java/com/onscreensync/tvapp/apiresponses/configs/DisplayApiConfig.kt
package com.onscreensync.tvapp.apiresponses.configs

import com.google.gson.annotations.SerializedName

data class DisplayApiConfig(
    @SerializedName("base-endpoint")
    val baseEndpoint: String?,
    @SerializedName("device-code-url")
    val deviceCodeUrl: String?,
    @SerializedName("device-info-url")
    val deviceInfoUrl: String?,
    @SerializedName("device-token-request-url")
    val deviceTokenRequestUrl: String?,
    @SerializedName("device-refresh-token-request-url")
    val deviceRefreshTokenRequestUrl: String?,
    @SerializedName("content-data-url")
    val contentDataUrl: String?,
    @SerializedName("signalr-negotiation-url")
    val signalrNegotiationUrl: String?,
    @SerializedName("signalr-add-connection-url")
    val signalrAddConnectionUrl: String?,
    @SerializedName("signalr-remove-connection-url")
    val signalrRemoveConnectionUrl: String?,
    @SerializedName("messages")
    val configMessages: DisplayApiConfigMessages?
)
